core/domain/collection_services.py

```diff
@@ -574,6 +576,29 @@ def get_collection_snapshots_metadata(collection_id):
         collection_id, version_nums)
 
 
+def _get_contributor_ids_for_collection(col_id):
```

Comment:
Yes! I'm making some clean ups to the design doc, I just think that some of my tests actually weren't specced out fully. I'll let you take a look at the cases I've covered once I push that up, totally open to doing more though I *think* I have covered/will cover the relevant issues.